client/client.go

```diff
@@ -123,6 +143,29 @@ func (c *Client) doRequest(req *http.Request) ([]byte, int, error) {
 	return body, res.StatusCode, err
 }
 
+func (c *Client) authenticate() error {
+	mapping, encodedData := map[string]string{"password": c.AdminPassword}, new(bytes.Buffer)
+	json.NewEncoder(encodedData).Encode(mapping)
+
+	resp, err := c.HTTPClient.Post(fmt.Sprintf("%s/users/%s/tokens", c.HostURL, c.AdminEmail), "application/json", encodedData)
```

Comment:
maybe introduce function for fmt.Sprintf("%s/users/%s/tokens", c.HostURL, c.AdminEmail)

like 
```
func apiLogin(host, user) {
    return fmt.Sprintf("%s/users/%s/tokens", host, user)
}
```
we are doing it for most of endpoints like:
https://github.com/shipa-corp/terraform-provider-shipa/blob/5fdbd65d2baa35ceda0d83825e61ec2e88259f21/client/client.go#L29-L51